### PR TITLE
[Examples] Add register_file

### PR DESCRIPTION
**Pull request by leonardt**

### CODE REVIEW

File: examples/tests/test_register_file.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
+import random
+import pytest
```
Comment (rsetaluri) [inline examples/tests/test_register_file.py:2]: nit
```suggestion
import pytest
import random
```

File: examples/tests/test_register_file.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+from ..register_file import RegisterFile
```
Comment (rsetaluri) [inline examples/tests/test_register_file.py:4]: any chance there's a better way to do this?

File: examples/tests/test_register_file.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+import magma as m
+import fault as f
+from hwtypes import BitVector
```
Comment (rsetaluri) [inline examples/tests/test_register_file.py:8]: nit
```suggestion
import fault as f
from hwtypes import BitVector
import magma as m
```

**Review by rsetaluri — APPROVED** [review]

File: examples/tests/test_register_file.py (hunk 2 above)
Comment (leonardt) [reply examples/tests/test_register_file.py:4]: I'm going to change the examples format to be in the form a jupyter notebook, which would put the test code in the same "file" and avoid the import issue (also making it easier to associate the test/design code in one place).

**Pull request merged**